src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -50,6 +52,54 @@ export default class SimidMapCreative extends BaseSimidCreative {
    * @private 
   */
   grantLocationAccess_() {
-    //ToDo(kristenmason@): implement map
+    //ToDo(kristenmason@): implement the Google Maps request access functionality
+    try{
+      this.simidProtocol.sendMessage(CreativeMessage.REQUEST_PAUSE).then(() => {
+        findNearest.classList.add("hidden");})
+    }
+    catch(exception){
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.PAUSE_NOT_HONORED, 
+        message: "Request to pause ad not honored"});
+    }
+    this.createButtonsMapState_();
+  }
+
+  /**
+   * Creates the Skip To Content and Return To Ad buttons once the user
+   *   grants permission to access their location and the map appears.
+   * @private 
+  */
+  createButtonsMapState_() {
+    const returnToAdButton = document.createElement("button");
+    returnToAdButton.innerHTML = "Return To Ad";
```

Comment:
use returnToAdButton.textContent here instead of innerHTML

Reply:
changed

Reply:
Nice catch Kaleab!

Try to avoid innerHTML whenever possible.  innerHTML will take whatever you put in as literal HTML, so if you were to for example run the following lines:
```
div.innerHTML = "<div onclick="functionToAttackComputer()">Text</div>";
```
That would be rendered as html, meaning that a click on that div would execute that function.
If you use textContent instead, you would see that as a literal string on the page instead.
In this instance, it probably doesn't matter since you are hardcoding "Return To Ad", but if this came from AdParameters fro example, then it would be very easy for an attacker to inject arbitrary code into your ad, so its good practice to only use innerHTML if you have a very specific need for it